sdks/java/io/hbase/src/main/java/org/apache/beam/sdk/io/hbase/HBaseIO.java

```diff
@@ -0,0 +1,681 @@
+     * the HBase instance, and a {@link HBaseIO.Read#withTableId tableId} that
+     * specifies which table to read. A {@link Filter} may also optionally be specified using
+     * {@link HBaseIO.Read#withFilter}.
+     */
+    @Experimental
+    public static Read read() {
+        return new Read(null, "", new SerializableScan(new Scan()));
+    }
+
+    /**
+     * A {@link PTransform} that reads from HBase. See the class-level Javadoc on
+     * {@link HBaseIO} for more information.
+     *
+     * @see HBaseIO
+     */
+    public static class Read extends PTransform<PBegin, PCollection<Result>> {
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read from the HBase instance
+         * indicated by the given configuration.
+         */
+        public Read withConfiguration(Configuration configuration) {
+            checkNotNull(configuration, "conf");
+            return new Read(new SerializableConfiguration(configuration),
+                    tableId, serializableScan);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read from the specified table.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withTableId(String tableId) {
+            checkNotNull(tableId, "tableId");
+            return new Read(serializableConfiguration, tableId, serializableScan);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will filter the rows read from HBase
+         * using the given scan.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withScan(Scan scan) {
+            checkNotNull(scan, "scan");
+            return new Read(serializableConfiguration, tableId, new SerializableScan(scan));
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will filter the rows read from HBase
+         * using the given row filter.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withFilter(Filter filter) {
+            checkNotNull(filter, "filter");
+            return withScan(serializableScan.getScan().setFilter(filter));
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read only rows in the specified range.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withKeyRange(ByteKeyRange keyRange) {
+            checkNotNull(keyRange, "keyRange");
+            byte[] startRow = keyRange.getStartKey().getBytes();
+            byte[] stopRow = keyRange.getEndKey().getBytes();
+            return withKeyRange(startRow, stopRow);
+        }
+
+        /**
+         * Returns a new {@link HBaseIO.Read} that will read only rows in the specified range.
+         *
+         * <p>Does not modify this object.
+         */
+        public Read withKeyRange(byte[] startRow, byte[] stopRow) {
+            checkNotNull(startRow, "startRow");
```

Comment:
ok, I did this by making those into a KeyRange and making the scan from the KeyRange now